contracts/YieldNotionalLever.sol

```diff
@@ -0,0 +1,436 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.14;
+
+import "@yield-protocol/utils-v2/contracts/cast/CastU128I128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastI128U128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256U128.sol";
+import "@yield-protocol/utils-v2/contracts/cast/CastU256I256.sol";
+import "@yield-protocol/vault-v2/other/notional/ERC1155.sol";
+import "./YieldLeverBase.sol";
+import "./NotionalTypes.sol";
+
+/// @title A contract to help users build levered position on notional
+/// @author iamsahu
+/// @notice Each external function has the details on how this works
+contract YieldNotionalLever is YieldLeverBase, ERC1155TokenReceiver {
+    using TransferHelper for IERC20;
+    using TransferHelper for IFYToken;
+    using CastU128I128 for uint128;
+    using CastI128U128 for int128;
+    using CastU256U128 for uint256;
+    using CastU256I256 for uint256;
+    Notional constant notional =
+        Notional(0x1344A36A1B56144C3Bc62E7757377D288fDE0369);
+    bytes6 constant ASSET_ID_MASK = 0xFFFF00000000;
+
+    /// @notice Struct to store fcash related information for an ilk
+    /// @param join The join of the underlying asset
+    /// @param maturity The maturity date of fCash
+    /// @param currencyId The id used for the fCash in notional
+    /// @dev We choose to store join redundantly as it is cheaper to load
+    ///     than to use an if statement to choose based on the currency id
+    struct IlkInfo {
+        FlashJoin join;
+        uint40 maturity;
+        uint16 currencyId;
+    }
+
+    /// @notice stores the information of the enabled ilks on the lever
+    mapping(bytes6 => IlkInfo) public ilkInfo;
+
+    constructor(Giver giver_) YieldLeverBase(giver_) {
+        notional.setApprovalForAll(address(ladle), true);
+    }
+
+    // TODO: Make it auth controlled when deploying
+    /// @notice Stores information regarding ilks (t) which are enabled on the lever
+    /// @param ilkId the id of the ilk
+    /// @param ilkInfo_ fCash related information for the ilk to being added
+    function setIlkInfo(bytes6 ilkId, IlkInfo calldata ilkInfo_) external {
```

Comment:
`ilkInfo` can be derived from `ilkId`. You would pull the join from the ladle, then `fCashId` from the join, and then `maturity` and `currencyId` from the `fCashId`.

Since the only input is `ilkId`, only fCash tenors that we have registered in the ladle can be recovered. You can have `invest` check if an `ilkId` is registered, and if not register it then.

The default would be to allow all fCash tenors in the ladle, and we wouldn't need governance in this lever.

<img width="344" alt="image" src="https://user-images.githubusercontent.com/38806121/202797386-767ba126-ae95-476b-b2fa-95f4f373eaa8.png">